index.js

```diff
@@ -517,8 +522,30 @@ function refFinder (ref, schema, externalSchema) {
       }
     }
   }
-  const result = (new Function('schema', code))(schema)
+  var result
+  try {
+    result = (new Function('schema', code))(schema)
+  } catch (err) {}
+
+  if (result === undefined) {
+    findBadKey(schema, walk.slice(1))
```

Comment:
Why this sentence is not in the previous `catch`?

Reply:
Because there is a case where the function above does not throw and just returns `undefined`.

Eg:
```js
schema['bad']['good'] // throws
schema['good']['bad'] // undefined and throws later
```